### ISSUE TITLE
In Tiles using terrain set mode "Match Side " DOES NOT correspond to 2x2 bitmap mode

### ISSUE THREAD

**Issue by LordBrkica** [godotengine/godot#79411]
### Godot version

4.1

### System information

Windows 11

### Issue description

In the official Godot 4.1 [documentation ](https://docs.godotengine.org/en/stable/tutorials/2d/using_tilesets.html#creating-terrain-sets-autotiling) it states that:
**
> Each terrain set is assigned a mode from Match Corners and Sides, Match Corners and Match sides. 

> Note
> The above modes correspond to the previous bitmask modes autotiles used in Godot 3.x: 2×2, 3×3 or 3×3 minimal.

** 

This is not true, **Match sides** mode DOES NOT correspond to a former 2×2 mode. This is a completely new mode without ANY documentation what so ever.

If the desired behaviour is not a bug then it would be nice if the official documentation at last states that this is mode breaks the compatibility with the old 2x2 mode. if we would be given a single example of using the new **Match sides** mode.

Right now it seems to me a vast number of existing assets is not easily usable with the new mode and I fail to see why was there a need to introduce such breaking changes without leaving the option to use the "normal" 2x2 mode.

### Steps to reproduce

Open Godot 3.5 Tileset and use 2x2 bitmap mode.

Than open Godot 4.1 and use the Match sides mode.

### Minimal reproduction project

N/A

**Comment by AThousandShips** [comment]
It doesn't say that though, "Match Corners and Sides" is meant to match it, does it not?

**Comment by dandeliondino** [comment]
I believe the statement in the docs may have been taken from an early blog post about the Godot 4 terrains modes. I don't know if it was intended to imply direct correlation, and I've seen a number of users get confused by it.

I spent some time clarifying the difference in the in-editor documentation for my Godot 4 bitmasking plugin TileBitTools, and I think it would be more helpful to make the actual Godot 4 documentation more accurate and straightforward. 

Godot 3 -> Godot 4

2x2 -> Corners

3x3 minimal (47-tile reduced set of 3x3 that matches all adjacent corners together) -> Corners and Sides

3x3 16-tile (not an official mode in Godot 3, a 16-tile reduced set of 3x3 that matches only sides together, created by placing ignore bits on the corners, and popularized with a template provided in Godot 3 docs) -> Sides

3x3 (full 256-tile mode that matches individual diagonal corners together) - not supported in Godot 4, but will be if PR #72030 is merged. It is currently set up to be a draw mode, not a terrain mode, though, which could increase confusion on this point.

**Comment by LordBrkica** [comment]
> Godot 3 -> Godot 4
> 
> 2x2 -> Corners

This is wrong, nothing in 4.1 corresponds to the 2x2 bitmap mod that was present in 3.

I am sad that the official documentation implies otherwise, if anything they should explicitly state that 2x2 bitmap mode is NOT supported.

**Comment by LordBrkica** [comment]
You are right there exists a corners mode - but you cannot do the same 2x2 bitmaping as its not a 2x2 bitmap structure but rather 2x2 +1 ??

Are you telling me 2x2 == 2x2+1  ?

1. Documentation should clearly state this....
2. How am I supposed to know what to do with the middle tile ?

So If we look at your image (with 2x2) and I want to treat now the new map as the old one - what am I (by default) to do with the middle tile ?

**Comment by AThousandShips** [comment]
Corresponds ≠ equals

You're nitpicking at this point

**Comment by Calinou** [comment]
@dandeliondino Please consider opening a pull request on https://github.com/godotengine/godot-docs :slightly_smiling_face:

**Comment by dandeliondino** [comment]
@LordBrkica 

I just opened a PR (https://github.com/godotengine/godot-docs/pull/7789) that should address this issue. Please let me know if there is still documentation missing on the points you brought up. You can preview the updated "Using TileSets" page [here](https://github.com/godotengine/godot-docs/blob/3af511222f68fa6ea6a9606234d5d8799411ca5a/tutorials/2d/using_tilesets.rst).
